google-cloud-spanner/src/main/java/com/google/cloud/spanner/BuiltInMetricsConstant.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import com.google.common.collect.ImmutableSet;
+import io.opentelemetry.api.common.AttributeKey;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class BuiltInMetricsConstant {
+
+  public static final String METER_NAME = "spanner.googleapis.com/internal/client";
+
+  public static final String GAX_METER_NAME = "gax-java";
+
+  static final String OPERATION_LATENCIES_NAME = "operation_latencies";
+  static final String ATTEMPT_LATENCIES_NAME = "attempt_latencies";
+  static final String OPERATION_LATENCY_NAME = "operation_latency";
+  static final String ATTEMPT_LATENCY_NAME = "attempt_latency";
+  static final String OPERATION_COUNT_NAME = "operation_count";
+  static final String ATTEMPT_COUNT_NAME = "attempt_count";
+
+  public static final Set<String> SPANNER_METRICS =
+      ImmutableSet.of(
+              OPERATION_LATENCIES_NAME,
+              ATTEMPT_LATENCIES_NAME,
+              OPERATION_COUNT_NAME,
+              ATTEMPT_COUNT_NAME)
+          .stream()
+          .map(m -> METER_NAME + '/' + m)
+          .collect(Collectors.toSet());
+
+  public static final String SPANNER_RESOURCE_TYPE = "spanner_instance_client";
+
+  public static final AttributeKey<String> PROJECT_ID_KEY = AttributeKey.stringKey("project_id");
+  public static final AttributeKey<String> INSTANCE_ID_KEY = AttributeKey.stringKey("instance_id");
+  public static final AttributeKey<String> LOCATION_ID_KEY = AttributeKey.stringKey("location");
+  public static final AttributeKey<String> INSTANCE_CONFIG_ID_KEY =
+      AttributeKey.stringKey("instance_config");
+
+  // These metric labels will be promoted to the spanner monitored resource fields
+  public static final Set<AttributeKey<String>> SPANNER_PROMOTED_RESOURCE_LABELS =
+      ImmutableSet.of(PROJECT_ID_KEY, INSTANCE_ID_KEY, INSTANCE_CONFIG_ID_KEY, LOCATION_ID_KEY);
```

Comment:
nit, iirc the `project_id` MonitoredResource label must always match the project the metrics are being written to, and so it is unnecessary to send it. We've taken this approach in our exporters since there is no benefit in sending it